src/twisted/internet/asyncfs.py

```diff
@@ -0,0 +1,287 @@
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+# -*- test-case-name: twisted.internet.test.test_async -*-
+"""
+This module contains asynchronous filesystem access functions and the threads-based implementation
+"""
+
+from typing import Tuple, Iterable, Optional, Callable, cast, Any
+import os
+import os.path
+import threading
+
+import attr
+from zope.interface import implementer, providedBy
+
+from twisted.internet import reactor
+from twisted.python.filepath import FilePath
+from twisted.internet.interfaces import (
+    IPushProducer,
+    IReactorFileAsync,
+    FileAsyncFlags,
+    IProducer,
+    IConsumer,
+    IAsyncReader,
+    IAsyncWriter,
+    DEFAULT_CHUNK_SIZE,
+    DEFAULT_BUFFER_MAX,
+)
+from twisted.python.threadpool import ThreadPool
+from twisted.internet.threads import (
+    deferToThreadPool,
+    blockingCallFromThread,
+)
+from twisted.internet.defer import Deferred
+from twisted.logger import Logger
+
+log = Logger()
+
+
+def openAsync(
+    path: FilePath,
+    flags: FileAsyncFlags,
+    user_reactor: Any = None,
```

Comment:
`user_reactor` should be `userReactor`.  Or probably just `reactor`?  That is the conventional name.

Reply:
surely just `reactor`  would clash with `from twisted.internet import reactor`?
`userReactor` is fine

Reply:
Yea.  Maybe don't have a default at all and remove the global reactor import from the module entirely.

Reply:
`openAsync` is intended as a conveience function, it seems odd to require callers to provide the reactor every time.